aquarius/events/decryptor.py

```diff
@@ -17,18 +19,34 @@
 
 def decrypt_ddo(w3, provider_url, contract_address, chain_id, txid, hash, es_instance):
     aquarius_account = get_aquarius_wallet()
-    nonce = str(int(datetime.now(timezone.utc).timestamp()))
+    # get nonce
+    try:
```

Comment:
Also, I would have this nonce retrieval process as a separate function with the only parameters needed being the address and the provider_url. Please also add unit tests for this part.